azurelinuxagent/common/cgroupconfigurator.py

```diff
@@ -499,12 +551,13 @@ def check_cgroups(self, cgroup_metrics):
             except CGroupsException as exception:
                 errors.append(exception)
 
-            disable = not process_check_success and conf.get_cgroup_disable_on_process_check_failure() \
-                      or \
-                      not quota_check_success and conf.get_cgroup_disable_on_quota_check_failure()
+            reason = "Check on cgroups failed:\n{0}".format("\n".join([ustr(e) for e in errors]))
+
+            if not process_check_success and conf.get_cgroup_disable_on_process_check_failure():
+                self.disable(reason, DisableCgroups.ALL)
 
-            if disable:
-                self.disable("Check on cgroups failed:\n{0}".format("\n".join([ustr(e) for e in errors])))
+            if not quota_check_success and conf.get_cgroup_disable_on_quota_check_failure():
```

Comment:
Ideally it should be provided by extension and we should disable that particular extension if extension cpu throttled. 

`Shouldn't we disable extension cgroups too if we're unable to set the quota?
`
That's the good point. Not sure, I will bring this up when I work on quota changes. As of today, we are not even doing for agent. Above line of check only for agent cpu throttling.

Overall, the disabling extensions is not needed for monitoring since quotas not introduced here. so I will address all these when we are going to add quotas code changes.